gen/suites/actor_deletion/main.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	"github.com/filecoin-project/specs-actors/actors/builtin"
+	"github.com/filecoin-project/specs-actors/actors/runtime/exitcode"
+	"github.com/filecoin-project/test-vectors/chaos"
+	. "github.com/filecoin-project/test-vectors/gen/builders"
+	. "github.com/filecoin-project/test-vectors/schema"
+)
+
+func main() {
+	g := NewGenerator()
+	defer g.Wait()
+
+	g.MessageVectorGroup("no_beneficiary",
+		&MessageVectorGenItem{
+			Metadata: &Metadata{
+				ID:      "delete-w-zero-balance",
+				Version: "v1",
+				Desc:    "actor with zero balance is deleted, does not require beneficiary",
+			},
+			Selector: Selector{"chaos_actor": "true"},
+			Func:     deleteActor,
+		},
+	)
+
+	g.MessageVectorGroup("beneficiary",
+		&MessageVectorGenItem{
+			Metadata: &Metadata{
+				ID:      "delete-w-balance-and-beneficiary",
+				Version: "v1",
+				Desc:    "actor with non-zero balance is deleted and sends funds to beneficiary",
+			},
+			Selector: Selector{"chaos_actor": "true"},
+			Func:     deleteActorWithBeneficiary(big.NewInt(50), address.Undef, exitcode.Ok),
+		},
+		// FIXME: this currently panics
+		// &MessageVectorGenItem{
+		// 	Metadata: &Metadata{
+		// 		ID:      "fail-delete-w-balance-and-unkown-beneficiary",
+		// 		Version: "v1",
+		// 		Desc:    "fails when actor with non-zero balance is deleted but beneficiary address is unknown",
+		// 	},
+		// 	Selector: Selector{"chaos_actor": "true"},
+		// 	Func:     deleteActorWithBeneficiary(big.NewInt(50), MustNewSECP256K1Addr("!👹*_👹!"), exitcode.SysErrorIllegalActor),
+		// },
+		&MessageVectorGenItem{
+			Metadata: &Metadata{
+				ID:      "fail-delete-w-balance-and-self-beneficiary",
+				Version: "v1",
+				Desc:    "fails when actor with non-zero balance is deleted but beneficiary is the calling actor",
+				Comment: "should abort if the beneficiary is the calling actor, see https://github.com/filecoin-project/specs-actors/blob/bcd83e8eb0a98b684851e574a2bd8d4130e21a51/actors/runtime/runtime.go#L117",
```

Comment:
@zenground0 @anorth 🙏 can you confirm what the exit code _should_ be? My guess would be `SysErrorIllegalArgument` based on https://github.com/filecoin-project/specs-actors/blob/master/actors/runtime/exitcode/reserved.go#L48?

Reply:
`SysErrorIllegalArgument` sounds perfect

Reply:
https://github.com/filecoin-project/lotus/pull/3478